Update RESULTS_GUIDELINES.md

nv-rborkar opened the pull request:
Update includes:

1. Brief Introduction
2. Separate section to guide (Open vs Closed) comparisons
3. Prohibits comparison of Open vs Closed scores on different hardware systems
4. Rules for making non-standard comparisons based on “MLPerf” models
5. Process for handling violations, remedies & penalties

nv-rborkar commented:
@johntran-nv , @petermattson , @TheKanter : Please review

petermattson commented:
Hi Ritika,
I think this CL identifies some important issues we need to address.
However, I'm worried that the TOU is already so long that no one reads it.
:-) Hence, proposal in WG today to discuss more.
Best,
Peter

On Wed, Apr 28, 2021 at 8:55 AM Ritika Borkar ***@***.***>
wrote:

> @johntran-nv <https://github.com/johntran-nv> , @petermattson
> <https://github.com/petermattson> , @TheKanter
> <https://github.com/TheKanter> : Please review
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mlcommons/policies/pull/69#issuecomment-828570554>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIIVUHNUI76JOQR73COHAJTTLAVXHANCNFSM4XN6CXLQ>
> .
>

petermattson commented:
Thanks for moving this forward and for your patience with getting a full review! :-) Here are some suggestions we should probably discuss in WG. Also, we need to get any changes here reviewed by Inference+Training and approved by BoD. This is an org-wide, public affecting change. If we get aligned in WG, David and I can help with the further alignment.

// Footnotes are a PITA, suggest we keep them as simple as possible to encourage compliance
// Many details handled by requirement to include basics and ID differences in text
1. Any use of MLPerf results of must include the following in a footnote: benchmark suite, version, and division, MLPerf result ID (major-version.minor-version.entry.benchmark), and a clear reference to MLPerf trademark. For example:
_[1] MLPerf v0.5 Inference Closed, entry 0.5-12. MLPerf name and logo are trademarks. ._

// I'd favor original text here. People should be able to compare closed and open provided they are clear about what they are doing per differences requirement. Open is intended to show the benefits of innovation relative to closed baselines.

## When comparing MLPerf results, you must identify any benchmark differences
...
## When comparing MLPerf results of different divisions, one must identify any benchmark or dataset differences
...

// Suggest we strike this text. If a vendor chooses not to benchmark officially, unofficial measurements should not require their permission.
"If you cannot do that, you must get permission from the vendor."

// This section is unclear and long, suggest we strike in its entirety. Weird corner cases can be handled by "reserves sole right to determine" clause.
## Additional comparisons should provide context of MLPerf benchmark results
...

// Good to specify this, but would probably strive to make light weight when possible (many cases are just ignorance), and to be more transparent and leverage existing processes for more serious problems. Suggest that instead of special closed-door committee we have two stage process. (1) Can raise a complaint to ED + WG chairs of appropriate benchmark WG and ask to inform potential violator and request fix to usage. (2) If ED, chairs, objector, or potential violator are not able to reach mutually satisfactory conclusion, can raise in WG and request resolution through WG vote.
## Violation Determination, Remedies, and Penalties
...

nv-rborkar commented:
Thanks for the feedback Peter!

- Incorporated changes to keep the footnotes as is, assuming basic text includes the relevant data about submissions
- Updated language for unofficial measurements as suggested
- Updated the process for violation resolution

I have some questions regarding Open vs Closed text and the Additional Comparisons text, that we can discuss in the WG.

TheKanter commented:
My apologies folks. I have directly modified the results guidelines per some discussions with legal. Those changes may force rewriting some aspects of these proposed changes.

psyhtest commented:
Performance is clearly a primary metric. Is power a primary or a secondary metric? The way I read this, in either case, a derived metric such as inferences per second per Watt would need to be presented alongside the primary metrics. Is that correct?

nv-rborkar commented:
@psyhtest : Thats right, whenever doing a secondary metric comparison one must also present the primary metric (official score observed by MLPerf) comparison alongside.

s-idgunji commented:
Regarding Power -
We should not use any normalized (derived) metric because the metric is not constant: Performance and Power do not scale linearly, so trying to do a ratio is misleading. Performance and Power should continue to be reported separately as has been done in r1.0 Inference submissions.

In context of Primary/Secondary metric :  The discussion ongoing is that the primary metric is within a larger context of a Performance metric : so time to train is a primary perf metric for Training , and perhaps Inferences/sec for Inference. Power is orthogonal to Performance as a metric, so we can have our associated primary metric for Power. Watts where it is Inference/sec and Energy consumed where it is a time specific Performance metric like "latency" or "time to train".  Power is not a secondary metric , but an independent primary metric separate to Performance .

rnaidu02 commented:
Inference WG has no objections

nv-rborkar commented:
> My apologies folks. I have directly modified the results guidelines per some discussions with legal. Those changes may force rewriting some aspects of these proposed changes.
> 
> Can we ensure that when this is merged that any changes respect the pre-existing changes?

@TheKanter to resolve conflicts with your re-writes, I have created a new PR on top of latest master at #87

nv-rborkar commented:
@TheKanter can you please merge #87 and close this one, since the Inference WG has also approved this PR now.
